### DISCUSSION TITLE
Search table through a text field (called "the flexible way" in the docs) not working

### DISCUSSION THREAD

**Discussion by egattulli** [Budibase/budibase#10288] [Help] [answered]
Hi everyone,

I've been trying to create a screen where there is a table and a text field that functions as a search field. I followed step by step the docs guide (https://docs.budibase.com/docs/adding-a-search-field) but it does not work! I have the 2.5.2 version. Do you know whether there are bugs in this version?

Thanks a lot!

**Comment by ConorWebb96** [comment, marked as answer]
Hey @egattulli,

This should be quite a quick thing to implement using the [Table Block ](https://docs.budibase.com/docs/table-block) component.

To quickly set this up add a table block and connect it to the data you want the table to display.

**Table component - click to add**
<img width="1920" alt="tableBlock" src="https://user-images.githubusercontent.com/126772285/231741475-9b7a8485-fac0-4b05-874e-2c2605347420.png">

**Table block settings including select data-set and filtering select.**
<img width="305" alt="tableblock Settings" src="https://user-images.githubusercontent.com/126772285/231741683-7afd9cfb-4bbd-4679-b8cc-f56bae663ceb.png">

**The field i've selected for searching.**
<img width="300" alt="select search field" src="https://user-images.githubusercontent.com/126772285/231742075-acb31c1e-31cf-4479-bb3b-3b02ad6ee9ff.png">

**Result of selecting this search field.**
<img width="686" alt="result of select search" src="https://user-images.githubusercontent.com/126772285/231742225-fd891d8b-3b79-49a7-9523-a86650eb5732.png">

**Demo of the end product working**
![Searching Table Block working](https://user-images.githubusercontent.com/126772285/231742378-1d00bddd-12ca-4432-9b5a-136fbf501c90.gif)

A small note, if you want to see how this is working you can **eject** the table block and see how it is all setup.
<img width="354" alt="eject table block" src="https://user-images.githubusercontent.com/126772285/231742977-4501f5f8-79c1-428f-86a6-c339ebab7970.png">

Then from here you can examine the define filters which is how its doing the filtering.
<img width="1917" alt="data filtering" src="https://user-images.githubusercontent.com/126772285/231743244-b738abe3-ea8e-4307-85f4-e4858438a9bf.png">

I'm also running version 2.5.2, so I don't think this is a bug.

Hopefully this helps you with figuring this out.

**Reply by egattulli** [reply]
Thank you @ConorWebb96! After ejecting the Table Block I saw where the issue was. Instead of setting the filter as "Starts with" in the Data Provider, I set "Equals". This caused the table to return no corresponding rows when I typed something in the text field. Thanks again.

**Reply by ConorWebb96** [reply]
No worries @egattulli, glad I helped. Happy app building!
